### ISSUE TITLE
Fedora rpm build | two tests are failing

### ISSUE THREAD

**Issue by firefly-cpp** [NiaOrg/NiaPy#252]
Two tests are failing:
- test_Custom_works_fine
- test_griewank_works_fine

More info:

https://src.fedoraproject.org/rpms/python-niapy/blob/master/f/python-niapy.spec#_86

**Comment by sanjayankur31** [comment]
These tests are disabled in the Fedora build, but I can enable them and provide detailed info and the traceback.

Fedora 33 (rawhide) is currently on Python3.9. The versions of all the python packages that are involved can be see in the log file here:

https://kojipkgs.fedoraproject.org//packages/python-niapy/2.0.0/0.1rc10.fc33/data/logs/noarch/root.log

For example:

```
DEBUG util.py:623:  Installing:
DEBUG util.py:623:   python3-astroid                         noarch   2.4.2-1.git2d25e84.fc33        build  308 k
DEBUG util.py:623:   python3-devel                           ppc64le  3.9.0~b3-1.fc33                build  204 k
DEBUG util.py:623:   python3-matplotlib                      ppc64le  3.3.0-0.2.rc1.fc33             build  2.7 M
DEBUG util.py:623:   python3-numpy                           ppc64le  1:1.19.0-1.fc33                build  4.5 M
DEBUG util.py:623:   python3-pytest                          noarch   5.4.3-1.fc33                   build  1.4 M
DEBUG util.py:623:   python3-scipy                           ppc64le  1.5.0-1.fc33                   build   15 M
DEBUG util.py:623:   python3-setuptools                      noarch   47.3.1-1.fc33                  build  598 k
DEBUG util.py:623:   python3-sphinx                          noarch   1:3.1.1-1.fc33                 build  2.0 M
DEBUG util.py:623:   python3-sphinx_rtd_theme                noarch   0.4.3-13.fc33                  build  3.2 M
DEBUG util.py:623:   python3-xlsxwriter                      noarch   1.2.9-1.fc33                   build  228 k
DEBUG util.py:623:  Installing dependencies:
DEBUG util.py:623:   cairo                                   ppc64le  1.16.0-8.fc33                  build  800 k
DEBUG util.py:623:   dejavu-sans-fonts                       noarch   2.37-12.fc33                   build  1.3 M
DEBUG util.py:623:   fontawesome-fonts                       noarch   4.7.0-8.fc32                   build  205 k
DEBUG util.py:623:   fontawesome-fonts-web                   noarch   4.7.0-8.fc32                   build  421 k
DEBUG util.py:623:   fontconfig                              ppc64le  2.13.92-11.fc33                build  295 k
DEBUG util.py:623:   fonts-filesystem                        noarch   1:2.0.5-3.fc33                 build  8.5 k
DEBUG util.py:623:   freetype                                ppc64le  2.10.1-2.fc32                  build  445 k
DEBUG util.py:623:   gdbm-libs                               ppc64le  1:1.18.1-3.fc32                build   58 k
DEBUG util.py:623:   glib2                                   ppc64le  2.65.0-2.fc33                  build  2.8 M
DEBUG util.py:623:   google-roboto-slab-fonts                noarch   1.100263-0.13.20150923git.fc33 build  241 k
DEBUG util.py:623:   jbigkit-libs                            ppc64le  2.1-18.fc32                    build   57 k
DEBUG util.py:623:   langpacks-core-font-en                  noarch   3.0-3.fc33                     build  9.4 k
DEBUG util.py:623:   lato-fonts                              noarch   2.015-9.fc32                   build  3.1 M
DEBUG util.py:623:   lcms2                                   ppc64le  2.11-1.fc33                    build  196 k
DEBUG util.py:623:   libX11                                  ppc64le  1.6.9-3.fc32                   build  716 k
DEBUG util.py:623:   libX11-common                           noarch   1.6.9-3.fc32                   build  153 k
DEBUG util.py:623:   libXau                                  ppc64le  1.0.9-3.fc32                   build   31 k
DEBUG util.py:623:   libXext                                 ppc64le  1.3.4-3.fc32                   build   41 k
DEBUG util.py:623:   libXft                                  ppc64le  2.3.3-3.fc32                   build   68 k
DEBUG util.py:623:   libXrender                              ppc64le  0.9.10-11.fc32                 build   29 k
DEBUG util.py:623:   libgfortran                             ppc64le  10.1.1-2.fc33                  build  805 k
DEBUG util.py:623:   libimagequant                           ppc64le  2.12.6-2.fc32                  build   68 k
DEBUG util.py:623:   libjpeg-turbo                           ppc64le  2.0.4-3.fc33                   build  206 k
DEBUG util.py:623:   libpng                                  ppc64le  2:1.6.37-3.fc32                build  137 k
DEBUG util.py:623:   libqhull                                ppc64le  1:7.2.1-3.fc32                 build  185 k
DEBUG util.py:623:   libquadmath                             ppc64le  10.1.1-2.fc33                  build  215 k
DEBUG util.py:623:   libtiff                                 ppc64le  4.1.0-2.fc32                   build  205 k
DEBUG util.py:623:   libwebp                                 ppc64le  1.1.0-3.fc33                   build  285 k
DEBUG util.py:623:   libxcb                                  ppc64le  1.13.1-4.fc32                  build  247 k
DEBUG util.py:623:   openblas                                ppc64le  0.3.9-3.fc33                   build   30 k
DEBUG util.py:623:   openblas-threads                        ppc64le  0.3.9-3.fc33                   build  3.1 M
DEBUG util.py:623:   openjpeg2                               ppc64le  2.3.1-6.fc33                   build  181 k
DEBUG util.py:623:   pixman                                  ppc64le  0.40.0-1.fc33                  build  231 k
DEBUG util.py:623:   python-pip-wheel                        noarch   20.1.1-4.fc33                  build  1.2 M
DEBUG util.py:623:   python-rpm-macros                       noarch   3.9-4.fc33                     build   14 k
DEBUG util.py:623:   python-setuptools-wheel                 noarch   47.3.1-1.fc33                  build  294 k
DEBUG util.py:623:   python3                                 ppc64le  3.9.0~b3-1.fc33                build   26 k
DEBUG util.py:623:   python3-attrs                           noarch   19.3.0-4.fc33                  build   66 k
DEBUG util.py:623:   python3-babel                           noarch   2.8.0-6.fc33                   build  5.7 M
DEBUG util.py:623:   python3-cairo                           ppc64le  1.18.2-5.fc33                  build   99 k
DEBUG util.py:623:   python3-chardet                         noarch   3.0.4-17.fc33                  build  194 k
DEBUG util.py:623:   python3-cycler                          noarch   0.10.0-15.fc33                 build   19 k
DEBUG util.py:623:   python3-dateutil                        noarch   1:2.8.1-1.fc33                 build  291 k
DEBUG util.py:623:   python3-docutils                        noarch   0.16-2.fc33                    build  1.5 M
DEBUG util.py:623:   python3-idna                            noarch   2.9-2.fc33                     build   98 k
DEBUG util.py:623:   python3-imagesize                       noarch   1.2.0-1.fc33                   build   17 k
DEBUG util.py:623:   python3-jinja2                          noarch   2.11.2-5.fc33                  build  489 k
DEBUG util.py:623:   python3-kiwisolver                      ppc64le  1.2.0-2.fc33                   build   79 k
DEBUG util.py:623:   python3-lazy-object-proxy               ppc64le  1.5.0-1.fc33                   build   37 k
DEBUG util.py:623:   python3-libs                            ppc64le  3.9.0~b3-1.fc33                build  7.5 M
DEBUG util.py:623:   python3-markupsafe                      ppc64le  1.1.1-6.fc33                   build   33 k
DEBUG util.py:623:   python3-matplotlib-data                 noarch   3.3.0-0.2.rc1.fc33             build  1.8 M
DEBUG util.py:623:   python3-matplotlib-data-fonts           noarch   3.3.0-0.2.rc1.fc33             build  2.3 M
DEBUG util.py:623:   python3-matplotlib-tk                   ppc64le  3.3.0-0.2.rc1.fc33             build   37 k
DEBUG util.py:623:   python3-more-itertools                  noarch   7.2.0-5.fc33                   build   56 k
DEBUG util.py:623:   python3-numpy-f2py                      ppc64le  1:1.19.0-1.fc33                build  213 k
DEBUG util.py:623:   python3-olefile                         noarch   0.46-11.fc33                   build   56 k
DEBUG util.py:623:   python3-packaging                       noarch   20.4-1.fc33                    build   67 k
DEBUG util.py:623:   python3-pillow                          ppc64le  7.1.2-2.fc33                   build  711 k
DEBUG util.py:623:   python3-pluggy                          noarch   0.13.1-1.fc33                  build   38 k
DEBUG util.py:623:   python3-py                              noarch   1.9.0-1.fc33                   build  430 k
DEBUG util.py:623:   python3-pygments                        noarch   2.6.1-3.fc33                   build  1.8 M
DEBUG util.py:623:   python3-pyparsing                       noarch   2.4.7-3.fc33                   build  150 k
DEBUG util.py:623:   python3-pysocks                         noarch   1.7.1-6.fc33                   build   35 k
DEBUG util.py:623:   python3-pytz                            noarch   2020.1-2.fc33                  build   49 k
DEBUG util.py:623:   python3-requests                        noarch   2.23.0-4.fc33                  build  112 k
DEBUG util.py:623:   python3-rpm-generators                  noarch   11-7.fc33                      build   26 k
DEBUG util.py:623:   python3-rpm-macros                      noarch   3.9-4.fc33                     build   13 k
DEBUG util.py:623:   python3-six                             noarch   1.15.0-1.fc33                  build   36 k
DEBUG util.py:623:   python3-snowballstemmer                 noarch   1.9.0-6.fc33                   build  149 k
DEBUG util.py:623:   python3-sphinx-theme-alabaster          noarch   0.7.12-9.fc33                  build   24 k
DEBUG util.py:623:   python3-sphinxcontrib-applehelp         noarch   1.0.2-1.fc33                   build   34 k
DEBUG util.py:623:   python3-sphinxcontrib-devhelp           noarch   1.0.2-1.fc33                   build   29 k
DEBUG util.py:623:   python3-sphinxcontrib-htmlhelp          noarch   1.0.3-1.fc33                   build   37 k
DEBUG util.py:623:   python3-sphinxcontrib-jsmath            noarch   1.0.1-8.fc33                   build   16 k
DEBUG util.py:623:   python3-sphinxcontrib-qthelp            noarch   1.0.3-1.fc33                   build   33 k
DEBUG util.py:623:   python3-sphinxcontrib-serializinghtml   noarch   1.1.4-1.fc33                   build   32 k
DEBUG util.py:623:   python3-tkinter                         ppc64le  3.9.0~b3-1.fc33                build  309 k
DEBUG util.py:623:   python3-typed_ast                       ppc64le  1.4.1-3.fc33                   build  222 k
DEBUG util.py:623:   python3-urllib3                         noarch   1.25.8-3.fc33                  build  173 k
DEBUG util.py:623:   python3-wcwidth                         noarch   0.2.5-1.fc33                   build   41 k
DEBUG util.py:623:   python3-wrapt                           ppc64le  1.12.1-2.fc33                  build   54 k
DEBUG util.py:623:   tcl                                     ppc64le  1:8.6.10-2.fc33                build  1.2 M
DEBUG util.py:623:   tk                                      ppc64le  1:8.6.10-3.fc32                build  1.6 M
DEBUG util.py:623:   xml-common                              noarch   0.6.3-54.fc32                  build   31 k
```

**Comment by sanjayankur31** [comment]
Here's the complete trace:

```
+ pytest -ra
============================= test session starts ==============================
platform linux -- Python 3.9.0b3, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /builddir/build/BUILD/NiaPy-2.0.0rc10
collected 938 items
NiaPy/tests/test_aba.py ............                                     [  1%]
NiaPy/tests/test_abc.py ...........                                      [  2%]
NiaPy/tests/test_algorithm.py .....................                      [  4%]
NiaPy/tests/test_argparser.py .....                                      [  5%]
NiaPy/tests/test_aso.py ............................                     [  8%]
NiaPy/tests/test_ba.py ...........                                       [  9%]
NiaPy/tests/test_bea.py ...........                                      [ 10%]
NiaPy/tests/test_benchmark.py .....                                      [ 11%]
NiaPy/tests/test_benchmark_functions.py ................................ [ 14%]
.................                                                        [ 16%]
NiaPy/tests/test_ca.py ...........                                       [ 17%]
NiaPy/tests/test_cro.py ..........                                       [ 18%]
NiaPy/tests/test_cs.py ..........                                        [ 19%]
NiaPy/tests/test_cso.py ..........                                       [ 20%]
NiaPy/tests/test_de.py .......................F....F.............F....F  [ 25%]
NiaPy/tests/test_es.py ............................................      [ 30%]
NiaPy/tests/test_fa.py ...........                                       [ 31%]
NiaPy/tests/test_foa.py ...........                                      [ 32%]
NiaPy/tests/test_fpa.py ............                                     [ 34%]
NiaPy/tests/test_fss.py ..........                                       [ 35%]
NiaPy/tests/test_fwa.py ..................................               [ 38%]
NiaPy/tests/test_ga.py ......................                            [ 41%]
NiaPy/tests/test_gsa.py ..........                                       [ 42%]
NiaPy/tests/test_gso.py ............................                     [ 45%]
NiaPy/tests/test_gwo.py ..........                                       [ 46%]
NiaPy/tests/test_hba.py ..........                                       [ 47%]
NiaPy/tests/test_hc.py ..........                                        [ 48%]
NiaPy/tests/test_hde.py ................................................ [ 53%]
....                                                                     [ 53%]
NiaPy/tests/test_hs.py ..................                                [ 55%]
NiaPy/tests/test_hsaba.py ...........                                    [ 57%]
NiaPy/tests/test_iter_gen_counters.py .......                            [ 57%]
NiaPy/tests/test_jade.py ...........                                     [ 58%]
NiaPy/tests/test_jde.py .................ss...........ss                 [ 62%]
NiaPy/tests/test_kh.py .......................................           [ 66%]
NiaPy/tests/test_mbo.py ...........                                      [ 67%]
NiaPy/tests/test_mfo.py ..........                                       [ 68%]
NiaPy/tests/test_mke.py .......................                          [ 71%]
NiaPy/tests/test_mts.py .................                                [ 73%]
NiaPy/tests/test_nmm.py .............                                    [ 74%]
NiaPy/tests/test_pso.py ................................................ [ 79%]
....................                                                     [ 81%]
NiaPy/tests/test_runner.py ...                                           [ 81%]
NiaPy/tests/test_sa.py .............                                     [ 83%]
NiaPy/tests/test_saba.py ..........                                      [ 84%]
NiaPy/tests/test_sade.py ........ss....ss                                [ 86%]
NiaPy/tests/test_sca.py ...........                                      [ 87%]
NiaPy/tests/test_task.py .....................................           [ 91%]
NiaPy/tests/test_ts.py ........ss                                        [ 92%]
NiaPy/tests/test_utility.py ............................................ [ 97%]
............................                                             [100%]
=================================== FAILURES ===================================
_____________________ ANpDETestCase.test_Custom_works_fine _____________________
self = <NiaPy.tests.test_de.ANpDETestCase testMethod=test_Custom_works_fine>
    def test_Custom_works_fine(self):
    	de_custom = self.algo(NP=40, F=0.5, CR=0.9, seed=self.seed)
    	de_customc = self.algo(NP=40, F=0.5, CR=0.9, seed=self.seed)
>   	AlgorithmTestCase.test_algorithm_run(self, de_custom, de_customc, MyBenchmark())
NiaPy/tests/test_de.py:85: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
NiaPy/tests/test_algorithm.py:321: in test_algorithm_run
    self.assertFalse(a.bad_run() or b.bad_run(), "Something went wrong at runtime of the algorithm --> %s" % a.exception)
E   AssertionError: True is not false : Something went wrong at runtime of the algorithm --> cannot convert float NaN to integer
____________________ ANpDETestCase.test_griewank_works_fine ____________________
self = <NiaPy.tests.test_de.ANpDETestCase testMethod=test_griewank_works_fine>
    def test_griewank_works_fine(self):
    	de_griewank = self.algo(NP=10, CR=0.5, F=0.9, seed=self.seed)
    	de_griewankc = self.algo(NP=10, CR=0.5, F=0.9, seed=self.seed)
>   	AlgorithmTestCase.test_algorithm_run(self, de_griewank, de_griewankc, 'griewank')
NiaPy/tests/test_de.py:90: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
NiaPy/tests/test_algorithm.py:321: in test_algorithm_run
    self.assertFalse(a.bad_run() or b.bad_run(), "Something went wrong at runtime of the algorithm --> %s" % a.exception)
E   AssertionError: True is not false : Something went wrong at runtime of the algorithm --> cannot convert float NaN to integer
____________________ ANpMsDETestCase.test_Custom_works_fine ____________________
self = <NiaPy.tests.test_de.ANpMsDETestCase testMethod=test_Custom_works_fine>
    def test_Custom_works_fine(self):
    	de_custom = self.algo(NP=40, F=0.5, CR=0.9, seed=self.seed)
    	de_customc = self.algo(NP=40, F=0.5, CR=0.9, seed=self.seed)
>   	AlgorithmTestCase.test_algorithm_run(self, de_custom, de_customc, MyBenchmark())
NiaPy/tests/test_de.py:141: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
NiaPy/tests/test_algorithm.py:321: in test_algorithm_run
    self.assertFalse(a.bad_run() or b.bad_run(), "Something went wrong at runtime of the algorithm --> %s" % a.exception)
E   AssertionError: True is not false : Something went wrong at runtime of the algorithm --> cannot convert float NaN to integer
___________________ ANpMsDETestCase.test_griewank_works_fine ___________________
self = <NiaPy.tests.test_de.ANpMsDETestCase testMethod=test_griewank_works_fine>
    def test_griewank_works_fine(self):
    	de_griewank = self.algo(NP=10, CR=0.5, F=0.9, seed=self.seed)
    	de_griewankc = self.algo(NP=10, CR=0.5, F=0.9, seed=self.seed)
>   	AlgorithmTestCase.test_algorithm_run(self, de_griewank, de_griewankc, 'griewank')
NiaPy/tests/test_de.py:146: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
NiaPy/tests/test_algorithm.py:321: in test_algorithm_run
    self.assertFalse(a.bad_run() or b.bad_run(), "Something went wrong at runtime of the algorithm --> %s" % a.exception)
E   AssertionError: True is not false : Something went wrong at runtime of the algorithm --> cannot convert float NaN to integer
=============================== warnings summary ===============================
NiaPy/tests/test_aso.py::ASOElitismTestCase::test_custom_works_fine
NiaPy/tests/test_aso.py::ASOElitismTestCase::test_griewank_works_fine
NiaPy/tests/test_fa.py::FATestCase::test_griewank_works_fine
NiaPy/tests/test_fa.py::FATestCase::test_works_fine
NiaPy/tests/test_iter_gen_counters.py::FATestCase::test_FA_evals_fine
NiaPy/tests/test_iter_gen_counters.py::FATestCase::test_FA_iters_fine
NiaPy/tests/test_jade.py::CrossRandCurr2pbestTestCase::test_function_fine
NiaPy/tests/test_runner.py::RunnerTestCase::test_runner_works_fine
  /usr/lib64/python3.9/site-packages/numpy/core/_asarray.py:83: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
    return array(a, dtype, copy=False, order=order)
NiaPy/tests/test_benchmark_functions.py::TestBenchmarkFunctions::test_infinity
  /builddir/build/BUILD/NiaPy-2.0.0rc10/NiaPy/benchmarks/infinity.py:84: RuntimeWarning: divide by zero encountered in double_scalars
    for i in range(D): val += X[i] ** 6 * (sin(1 / X[i]) + 2)
NiaPy/tests/test_benchmark_functions.py::TestBenchmarkFunctions::test_infinity
  /builddir/build/BUILD/NiaPy-2.0.0rc10/NiaPy/benchmarks/infinity.py:84: RuntimeWarning: invalid value encountered in sin
    for i in range(D): val += X[i] ** 6 * (sin(1 / X[i]) + 2)
NiaPy/tests/test_de.py::ANpDETestCase::test_Custom_works_fine
NiaPy/tests/test_de.py::ANpDETestCase::test_griewank_works_fine
NiaPy/tests/test_de.py::ANpMsDETestCase::test_Custom_works_fine
NiaPy/tests/test_de.py::ANpMsDETestCase::test_griewank_works_fine
  /builddir/build/BUILD/NiaPy-2.0.0rc10/NiaPy/algorithms/basic/de.py:564: RuntimeWarning: invalid value encountered in double_scalars
    return min(Lt_min + mu * avg / x_f, Lt_max)
NiaPy/tests/test_es.py: 16660 tests with warnings
  /builddir/build/BUILD/NiaPy-2.0.0rc10/NiaPy/algorithms/basic/es.py:293: DeprecationWarning: elementwise comparison failed; this will raise an error in the future.
    if e not in c: k += 1
NiaPy/tests/test_jade.py: 200 tests with warnings
  /usr/lib64/python3.9/site-packages/numpy/core/numeric.py:2378: DeprecationWarning: elementwise comparison failed; this will raise an error in the future.
    return bool(asarray(a1 == a2).all())
-- Docs: https://docs.pytest.org/en/latest/warnings.html
=========================== short test summary info ============================
SKIPPED [1] NiaPy/tests/test_jde.py:63: Not working
SKIPPED [1] NiaPy/tests/test_jde.py:69: Not working
SKIPPED [1] NiaPy/tests/test_jde.py:87: Not working
SKIPPED [1] NiaPy/tests/test_jde.py:93: Not working
SKIPPED [1] NiaPy/tests/test_sade.py:12: Not implementd jet!!!
SKIPPED [1] NiaPy/tests/test_sade.py:18: Not implementd jet!!!
SKIPPED [1] NiaPy/tests/test_sade.py:29: Not implementd jet!!!
SKIPPED [1] NiaPy/tests/test_sade.py:35: Not implementd jet!!!
SKIPPED [1] NiaPy/tests/test_ts.py:14: Not implemented jet!!!
SKIPPED [1] NiaPy/tests/test_ts.py:20: Not implemented jet!!!
FAILED NiaPy/tests/test_de.py::ANpDETestCase::test_Custom_works_fine - Assert...
FAILED NiaPy/tests/test_de.py::ANpDETestCase::test_griewank_works_fine - Asse...
FAILED NiaPy/tests/test_de.py::ANpMsDETestCase::test_Custom_works_fine - Asse...
FAILED NiaPy/tests/test_de.py::ANpMsDETestCase::test_griewank_works_fine - As...
==== 4 failed, 924 passed, 10 skipped, 16874 warnings in 814.69s (0:13:34) =====
```

Complete logs are attached:

- [niapy-rawhide-buildlog.txt](https://github.com/NiaOrg/NiaPy/files/4869422/niapy-rawhide-buildlog.txt)
- [niapy-rawhide-rootlog.txt](https://github.com/NiaOrg/NiaPy/files/4869423/niapy-rawhide-rootlog.txt)

**Comment by sisco0** [comment]
Have a good day, from what I could perceive, there would be an evaluation error at [https://github.com/NiaOrg/NiaPy/blob/d165b24c96a2afd34d600522a287a42ca4757775/NiaPy/tests/test_benchmark_functions.py#L438](test benchmark functions) because of the division in the _sin_ function at [https://github.com/NiaOrg/NiaPy/blob/d165b24c96a2afd34d600522a287a42ca4757775/NiaPy/benchmarks/infinity.py#L84](infinity benchmark).
The question is, why should we need to start with a zero-filled initialization at this benchmark? It would be recommended to do so with a 1e-6 value. Would not it?

**Comment by sisco0** [comment]
Maybe a self.algo.setParameters() call could fix some errors as well if inserted after
https://github.com/NiaOrg/NiaPy/blob/d165b24c96a2afd34d600522a287a42ca4757775/NiaPy/tests/test_de.py#L82

This would be tested.

**Comment by sisco0** [comment]
> Maybe a self.algo.setParameters() call could fix some errors as well if inserted after
> https://github.com/NiaOrg/NiaPy/blob/d165b24c96a2afd34d600522a287a42ca4757775/NiaPy/tests/test_de.py#L82
> 
> This would be tested.

Sorry, it seems that the error is at the proportional function call, where avg goes to infty and later, x_f goes to infinity as well.

**Comment by sisco0** [comment]
> > Maybe a self.algo.setParameters() call could fix some errors as well if inserted after
> > https://github.com/NiaOrg/NiaPy/blob/d165b24c96a2afd34d600522a287a42ca4757775/NiaPy/tests/test_de.py#L82
> > 
> > This would be tested.
> 
> Sorry, it seems that the error is at the proportional function call, where avg goes to infty and later, x_f goes to infinity as well.

This was fixed at #266 , please review it carefully to see if it is correct.

**Comment by sisco0** [comment]
Code styling reviewed and added a new pull request #267

**Comment by GregaVrbancic** [comment]
Thank you @sisco0! Issue solved by merging PR #267. 

@all-contributors add @sisco0 for code contribution
